File: browser/org/CreadoresProgram/ServerWebGamePost/client/ServerWebGamePostClient.js
class ServerWebGamePostClient {
    constructor(domain, port, isHttps) {
        this.domain = domain;
        this.port = port;
        this.isHttps = isHttps;
    }
    set processDatapacks(callback){
        return this.setProcessDatapacks(callback);
    }
    get processDatapacks(){
        return this.getProcessDatapacks();
    }
    setProcessDatapacks(callback){
        this.processDatapackspriv = callback;
    }
    getProcessDatapacks(){
        return this.processDatapackspriv;
    }
    sendDataPacket(datapack){
        let prefix = "";
        if(this.isHttps){
            prefix = "https://";
        }else{
            prefix = "http://";
        }
        fetch(prefix+this.domain+":"+this.port+"/ServerWebGamePost", {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify(datapack)
        }).then((reponse) => reponse.json()).then((data)=>{
            this.processSubDatapackspriv(data);
        }).catch((error) =>{
            throw error;
        });
    }
    processSubDatapackspriv(datapacks){
        for(let i in datapacks.datapacksLot){
            this.processDatapackspriv(datapacks.datapacksLot[i]);
        }
    }
}
Object.freeze(ServerWebGamePostClient.prototype);

File: LuaRoblox/org/CreadoresProgram/ServerWebGamePost/client/ServerWebGamePostClient.lua
local HttpService = game:GetService("HttpService")

local ServerWebGamePostClient = {}
ServerWebGamePostClient.__index = ServerWebGamePostClient

function ServerWebGamePostClient.new(domain, port, isHttps)
    local self = setmetatable({}, ServerWebGamePostClient)
    self.domain = domain
    self.port = port
    self.isHttps = isHttps
    self.processDatapackspriv = nil
    return self
end

function ServerWebGamePostClient:setProcessDatapacks(callback)
    self.processDatapackspriv = callback
end

function ServerWebGamePostClient:getProcessDatapacks()
    return self.processDatapackspriv
end

function ServerWebGamePostClient:sendDataPacket(datapack)
    local prefix = self.isHttps and "https://" or "http://"
    local url = prefix .. self.domain .. ":" .. tostring(self.port) .. "/ServerWebGamePost"
    local jsonData = HttpService:JSONEncode(datapack)
    
    local success, response = pcall(function()
        return HttpService:RequestAsync({
            Url = url,
            Method = "POST",
            Headers = {
                ["Content-Type"] = "application/json"
            },
            Body = jsonData
        })
    end)
    
    if success then
        local data = HttpService:JSONDecode(response.Body)
        self:processSubDatapackspriv(data)
    else
        error("HTTP Request failed: " .. tostring(response))
    end
end

function ServerWebGamePostClient:processSubDatapackspriv(datapacks)
    if datapacks and datapacks.datapacksLot then
        for _, dp in ipairs(datapacks.datapacksLot) do
            if self.processDatapackspriv then
                self.processDatapackspriv(dp)
            end
        end
    end
end

return ServerWebGamePostClient